store/generated_key/eigenda/eigenda.go

```diff
@@ -71,7 +75,28 @@ func (e Store) Put(ctx context.Context, value []byte) ([]byte, error) {
 	}
 
 	dispersalStart := time.Now()
-	blobInfo, err := e.client.PutBlob(ctx, value)
+
+	// We attempt to disperse the blob to EigenDA up to 3 times, unless we get a 400 error on any attempt.
+	blobInfo, err := retry.DoWithData(
+		func() (*disperser.BlobInfo, error) {
+			return e.client.PutBlob(ctx, value)
+		},
+		retry.RetryIf(func(err error) bool {
+			var errAPI api.ErrorAPI
+			if errors.As(err, &errAPI) {
+				if errAPI.ErrorFault() == api.ErrorFaultClient {
+					// we don't retry 400 errors because there is no point,
+					// we are passing invalid data
+					return false
+				}
+			}
+			return true
+		}),
+		// only return the last error. If it is a 503, then the handler will convert it to an http 503
+		// to signify to the client (batcher) to failover to ethda b/c eigenda is temporarily down.
+		retry.LastErrorOnly(true),
+		retry.Attempts(3),
```

Comment:
making it configurable?

Reply:
Done, see https://github.com/Layr-Labs/eigenda-proxy/pull/193/commits/873ab93cc96e59b89e203928490f310586baec95#diff-09e9e619c43538c83da73b9120ce0e0ae0f4bc06b2428dc0c2de9e5f0d5fe9d5R113